TheWarWithin/RogueSubtlety.lua

```diff
@@ -887,21 +889,34 @@ spec:RegisterAbilities( {
         end,
 
         handler = function ()
-            removeBuff( "honed_blades" )
-            applyDebuff( "target", "shadows_grasp", 8 )
 
-            if azerite.perforate.enabled and buff.perforate.up then
-                -- We'll assume we're attacking from behind if we've already put up Perforate once.
-                addStack( "perforate" )
-                gainChargeTime( "shadow_blades", 0.5 )
+            if talent.perforated_veins.enabled then
+                if buff.perforated_veins.stack < 4 then
+                    addStack( "perforated_veins" )
+                else removeBuff( "perforated_veins" )
```

Comment:
At 4 stacks, Perforated Veins is just consumed w/o other effect?  Need to determine if this is should be something like:

```
if talent.perforated_veins.enabled then
    if buff.perforated_veins_counter.stack < 4 then addStack( "perforated_veins" )
    else
        applyBuff( "perforated_veins" )
        removeBuff( "perforated_veins_counter" )
    end
end
```

Reply:
The counter is not for the current iteration. If it ever becomes referenced in the APL, would check if stacks = 4 during the cast. See other comment.